apps/cookbook/src/app/examples/chart-example/examples/chart-example-monthly-overview.component.ts

```diff
@@ -0,0 +1,142 @@
+  <kirby-chart [height]="height" [categories]="categories" [type]="'column'" [data]="adjustedMonthlyExpenseData" [options]="monthlyOverviewOptions"> </kirby-chart>
+  </kirby-card>
+  `,
+  codeSnippet: `
+  private monthlyOverviewClick: SeriesClickCallbackFunction = (ev: SeriesClickEventObject) => {
+    this.modalController.showAlert({
+      title: 'Clicked chart',
+      message: 'You clicked on column: ' + ev.point.category,
+      okBtnText: 'Ok',
+    });
+
+    this.selectedIdx = this.categories.indexOf(ev.point.category);
+    this.monthlyOverviewOptions.plotOptions.series.animation = false;
+    this.monthlyOverviewOptions.chart.events.load = colorPoints(this.selectedIdx);
+    this.monthlyOverviewOptions.chart.events.redraw = colorPoints(this.selectedIdx);
+    this.monthlyOverviewOptions = { ...this.monthlyOverviewOptions };
+  };
+  monthlyOverviewOptions: Options = {
+    chart: { events: {} },
+    plotOptions: {
+      series: {},
+      column: {
+        events: {
+          click: this.monthlyOverviewClick.bind(this),
+        },
+      },
+    },
+    yAxis: {
+      plotLines: [
+        {
+          width: 1,
+          color: getColor('semi-dark').value,
+          dashStyle: 'Dash',
+          value: 872,
+        },
+      ],
+    },
+  };
+  `,
+};
+
+@Component({
+  selector: config.selector,
+  template: config.template,
+})
+export class ChartExampleMonthlyOverviewComponent implements Example {
+  template: string = config.template;
+  codeSnippet: string = config.codeSnippet;
+  height = 150;
+
+  private categories = [
+    'mar',
+    'apr',
+    'maj',
+    'jun',
+    'jul',
+    'aug',
+    'sep',
+    'okt',
+    'nov',
+    'dec',
+    'jan',
+    'feb',
+  ];
+  private monthlyExpenseData = [0, 1400, 300, 500, 100, 1000, 1100, 450, 1350, 1200, 1250, 600];
+  private maxValue = Math.max(...this.monthlyExpenseData);
+  private selectedIdx = 0;
+
+  // lower limit is shown as 2% of max value for UX reasons
+  private lowerLimit = Math.max(...this.monthlyExpenseData) * 0.02;
+  adjustedMonthlyExpenseData = this.monthlyExpenseData.map((data) =>
+    this.lowerLimit >= data ? this.lowerLimit : data
+  );
+
+  private monthlyOverviewClick: SeriesClickCallbackFunction = (ev: SeriesClickEventObject) => {
+    this.modalController.showAlert({
+      title: 'Clicked chart',
+      message: 'You clicked on column: ' + ev.point.category,
+      okBtnText: 'Ok',
+    });
+
+    this.selectedIdx = this.categories.indexOf(ev.point.category);
+    this.monthlyOverviewOptions.plotOptions.series.animation = false;
+    this.monthlyOverviewOptions.chart.events.load = colorPoints(this.selectedIdx);
+    this.monthlyOverviewOptions.chart.events.redraw = colorPoints(this.selectedIdx);
+    this.monthlyOverviewOptions = { ...this.monthlyOverviewOptions };
+  };
+  monthlyOverviewOptions: Options = {
+    chart: { events: {} },
+    plotOptions: {
+      series: {},
+      column: {
```

Comment:
Why all this boilerplate? Should be removed

Reply:
Because event handler in example is accessing nested values of these objects